File: app/src/main/java/com/hearthappy/vma/model/testmodel/DynamicDTO.java
package com.hearthappy.vma.model.testmodel;


import java.io.Serializable;
import java.util.List;

public class DynamicDTO implements Serializable {
    public String c_time;
    public Integer comment_num;
    public String content;
    public String cover_img;
    public Integer id;
    public List<String> imgs;
    public Integer is_fork;
    public Integer is_give_like;
    public Integer is_show_location;
    public Integer popularity;
    public String proportion;
    public Integer support;
    public String type_name;
    public UserDTO user;
    public String topic_id = "";
    public String topic_title = "";
    public UserDynamicLocationDTO userDynamicLocation;
    public Integer user_id;
    public String video;
    public Integer is_attention;
    public Integer in_live;
    public String in_live_room_id;
    public int top_sort;
    public String head_portrait_box;
    public String year;
    public ResHotCommentList hot_comment;


    /**
     * //只在局部刷新的时候使用  0 不处理, 1 点赞刷新  2 评论数刷新
     */
    public Integer type = 0;

    public static class ResHotComment implements Serializable {
        public String c_time;
        public String content;
        public Integer id;
        public Integer is_author;
        public int is_give_like;
        public int support_num;
        public UserDTO user;
        public int user_dynamic_id;
        public int user_id;
    }

    public static class UserDTO implements Serializable {
        public String avatar;
        public Integer id;
        public Integer is_recommend;
        public Integer level;
        public String nickname;
        public String icon;
        public Integer sex;
        public int age;
        public Integer status;
        public String level_icon;
        public String charm_level_icon;
        public Integer charm_level;
        public Integer ifblack;//0:拉黑 1:未拉黑
        public NobilityUserBean nobility_user;

        public UserVipLevelDTO userVipLevel;


        public static class UserVipLevelDTO implements Serializable {
            public Integer experience;
            public String icon;
            public Integer id;
            public Integer level;
        }


        public static class NobilityUserBean implements Serializable {
            public String nobility_id;
            public String nobility_name;
            public String nobility_level;
            public String nobility_img;
            public Long end_time;
            public String end_time_format;
            public int nobility_img_switch;
        }

    }

    public static class UserDynamicLocationDTO implements Serializable {
        public Integer id;
        public String lat;
        public String lng;
        public String name;
        public Integer user_dynamic_id;
    }


    public void setIs_attention(Integer is_attention) {
        this.is_attention = is_attention;
    }

    public void setIs_give_like(Integer is_give_like) {
        this.is_give_like = is_give_like;
    }

    public void setSupport(Integer support) {
        this.support = support;
    }

    public void setComment_num(Integer comment_num) {
        this.comment_num = comment_num;
    }

    public void setType(Integer type) {
        this.type = type;
    }
}

File: app/src/main/java/com/hearthappy/vma/model/testmodel/DynamicDTO.kt
package com.hearthappy.vma.model.testmodel

import java.io.Serializable


class DynamicDTO : Serializable {
    var c_time: String? = null
    var comment_num: Int? = null
    var content: String? = null
    var cover_img: String? = null
    var id: Int? = null
    var imgs: List<String>? = null
    var is_fork: Int? = null
    var is_give_like: Int? = null
    var is_show_location: Int? = null
    var popularity: Int? = null
    var proportion: String? = null
    var support: Int? = null
    var type_name: String? = null
    var user: UserDTO? = null
    var topic_id: String = ""
    var topic_title: String = ""
    var userDynamicLocation: UserDynamicLocationDTO? = null
    var user_id: Int? = null
    var video: String? = null
    var is_attention: Int? = null
    var in_live: Int? = null
    var in_live_room_id: String? = null
    var top_sort: Int = 0
    var head_portrait_box: String? = null
    var year: String? = null
    var hot_comment: ResHotCommentList? = null


    /**
     * //只在局部刷新的时候使用  0 不处理, 1 点赞刷新  2 评论数刷新
     */
    var type: Int = 0

    class ResHotComment : Serializable {
        var c_time: String? = null
        var content: String? = null
        var id: Int? = null
        var is_author: Int? = null
        var is_give_like: Int = 0
        var support_num: Int = 0
        var user: UserDTO? = null
        var user_dynamic_id: Int = 0
        var user_id: Int = 0
    }

    class UserDTO : Serializable {
        var avatar: String? = null
        var id: Int? = null
        var is_recommend: Int? = null
        var level: Int? = null
        var nickname: String? = null
        var icon: String? = null
        var sex: Int? = null
        var age: Int = 0
        var status: Int? = null
        var level_icon: String? = null
        var charm_level_icon: String? = null
        var charm_level: Int? = null
        var ifblack: Int? = null //0:拉黑 1:未拉黑
        var nobility_user: NobilityUserBean? = null

        var userVipLevel: UserVipLevelDTO? = null


        class UserVipLevelDTO : Serializable {
            var experience: Int? = null
            var icon: String? = null
            var id: Int? = null
            var level: Int? = null
        }


        class NobilityUserBean : Serializable {
            var nobility_id: String? = null
            var nobility_name: String? = null
            var nobility_level: String? = null
            var nobility_img: String? = null
            var end_time: Long? = null
            var end_time_format: String? = null
            var nobility_img_switch: Int = 0
        }
    }

    class UserDynamicLocationDTO : Serializable {
        var id: Int? = null
        var lat: String? = null
        var lng: String? = null
        var name: String? = null
        var user_dynamic_id: Int? = null
    }

}